crates/rpc/src/method/get_compiled_casm.rs

```diff
@@ -0,0 +1,1549 @@
+                Self::ApplicationError(ApplicationError::ClassHashNotFound)
+            }
+            Error::Internal(e) => Self::InternalError(e),
+        }
+    }
+}
+
+/// Get the compiled casm for a given class hash.
+pub async fn get_compiled_casm(context: RpcContext, input: Input) -> Result<Output, Error> {
+    let span = tracing::Span::current();
+    let jh = tokio::task::spawn_blocking(move || -> Result<Output, Error> {
+        let _g = span.enter();
+
+        let mut db = context
+            .storage
+            .connection()
+            .context("Opening database connection")
+            .map_err(Error::Internal)?;
+
+        let tx = db
+            .transaction()
+            .context("Creating database transaction")
+            .map_err(Error::Internal)?;
+
+        // Get the class definition
+        let casm_definition = tx
+            .casm_definition(input.class_hash)
+            .context("Fetching class definition")
+            .map_err(Error::Internal)?
+            .ok_or(Error::ClassHashNotFound(input.class_hash))?;
+
+        // Convert to JSON string
+        let casm_definition_str = String::from_utf8_lossy(&casm_definition);
+
+        // Parse the casm definition
+        let casm_contract_class = CasmContractClass::try_from(casm_definition_str.as_ref())
+            .context("Parsing casm definition")
+            .map_err(|_| Error::CompilationFailed)?;
+
+        Ok(Output(casm_contract_class))
+    });
+
+    jh.await.context("Fetching compiled casm")?
+}
+
+#[cfg(test)]
+mod tests {
+
+    use pathfinder_common::casm_class::{
+        BigUintAsHex,
+        CasmContractEntryPoint,
+        CasmContractEntryPoints,
+    };
+    use pathfinder_common::{casm_hash, class_hash, felt, sierra_hash, EntryPoint};
+    use serde_json::json;
+    use starknet_gateway_test_fixtures::class_definitions::{
+        CAIRO_1_1_0_BALANCE_CASM_JSON,
+        CAIRO_1_1_0_BALANCE_SIERRA_JSON,
+    };
+    use tempfile::tempdir;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_get_casm() {
+        let rpc = setup().await.expect("RPC context");
+
+        // Test the happy path
+        let result = get_compiled_casm(rpc.clone(), input())
+            .await
+            .expect("result");
+        assert_eq!(result.0, expected());
+
+        // Test the not found path
+        let result = get_compiled_casm(rpc, input_not_found())
+            .await
+            .expect_err("result");
+        assert_matches::assert_matches!(result, Error::ClassHashNotFound(_));
+    }
+
+    async fn setup() -> anyhow::Result<RpcContext> {
+        let dir = tempdir().expect("tempdir");
+        let mut db_path = dir.path().to_path_buf();
+        db_path.push("db.sqlite");
+
+        let storage = pathfinder_storage::StorageBuilder::file(db_path)
+            .migrate()
+            .expect("storage")
+            .create_pool(std::num::NonZeroU32::new(1).expect("one"))
+            .expect("storage");
```

Comment:
```suggestion
        let storage = StorageBuilder::in_memory().unwrap();
```

Reply:
Thanks, used the other test as reference and didn't remember we had this!

Reply:
We should probably do the same to that other test.